Feature/map component

nbumbarger opened the pull request:
Adds functional map to the map placeholder, and connects it to the API to visualize the collection of requests (in the home view), individual requests (in the requests view), a user's requests (in the dashboard view), and individual tasks (in the task view).

nbumbarger committed:
Add mapbox-gl to dependencies

nbumbarger committed:
Merge branch 'develop' of https://github.com/hotosm/imagery-requests into feature/map-component

nbumbarger committed:
Add simple, unconnected map component

nbumbarger committed:
Add map component to all views

nbumbarger committed:
Fix map margins in dashboard view

nbumbarger committed:
Add geometry to feature/ featureclass util

nbumbarger committed:
Add features to map on dashboard view

nbumbarger committed:
Add feature display to dashboard view

nbumbarger committed:
Add feature display to map in task-page view

nbumbarger committed:
Move coordinate to feature conversion on map to parent views (dashboard and task page)

nbumbarger committed:
Add feature mapping to home view

nbumbarger committed:
ADd feature mapping to request view

nbumbarger committed:
Change name of TaskMap to DisplayMap

nbumbarger committed:
Temporarily link EditMap to home view for development, add temporariy CSS file

nbumbarger committed:
Scaffold EditMap

nbumbarger committed:
Add geojson editing layer

nbumbarger committed:
Move temporary edit map reference to edit page, to test interaction

nbumbarger committed:
Add style to map draw polygon

nbumbarger committed:
Limit drawing to one feature

nbumbarger committed:
Add listener to trash button to force switch to edit mode to enable editing when features cleared

Because mapboxgl-draw is not working correctly with mapbox, and most listeners are unavailable, we are using mapbox's map.on('click') listener to count active features. This doesn't fire when a button is clicked, so a listener needed to be added to the button to enable it after deletes

nbumbarger committed:
Refactor to make proper use of listeners

nbumbarger committed:
Remove changing edit mode in favor of CSS edit button disability

nbumbarger committed:
Remove temporary styles

nbumbarger commented:
This is ready. Most views include the view map, and the task page temporarily contains an example of the edit component.

nbumbarger committed:
Resolve merge conflicts with feature/request-form branch

nbumbarger committed:
Move Mapbox CSS import to index.html

nbumbarger committed:
Set map class in parent components by passing it as a property

nbumbarger committed:
Set draw control class using classList property

nbumbarger committed:
Remove deprecated CSS

nbumbarger committed:
Remove deprecated property passing in edit map

nbumbarger committed:
Move Mapbox token to config

nbumbarger committed:
Resolve merge conflict

danielfdsilva commented on app/assets/scripts/components/display-map.js:
```diff
@@ -0,0 +1,76 @@
+  propTypes: {
+    mapId: T.string,
+    results: T.object,
+    mapClass: T.string
```
Better to use `className` so it's a common property.

danielfdsilva commented on app/assets/scripts/components/display-map.js:
```diff
@@ -0,0 +1,76 @@
+  },
+
+  componentDidMount: function () {
+    mapboxgl.accessToken = config.mbToken;
```
move to `main.js` so we only do it once across the site. No need for duplication.

danielfdsilva commented on app/assets/scripts/components/edit-map.js:
```diff
@@ -0,0 +1,105 @@
+import { mbStyles } from '../utils/mapbox-styles';
+
+const EditMap = React.createClass({
+  displayName: 'DisplayMap',
```
Wrong `displayName`

danielfdsilva commented on app/assets/scripts/components/edit-map.js:
```diff
@@ -0,0 +1,105 @@
+
+  propTypes: {
+    mapId: T.string,
+    mapClass: T.string
```
Better to use `className` so it's a common property.

danielfdsilva commented on app/index.html:
```diff
@@ -43,13 +43,16 @@
-  
+
+    <link rel="stylesheet" href="/node_modules/mapbox-gl/dist/mapbox-gl.css">
+    <link rel="stylesheet" href="/node_modules/mapbox-gl-draw/dist/mapbox-gl-draw.css">
```
These need to be included in the build process above (line 43-45).

danielfdsilva reviewed (changes requested):
Other notes:
- Please add the token to `production.js` This is not sensitive and unless we go with a different map, won't change. (update the readme to reflect this)

nbumbarger committed:
Fix several small issues

Moves Mapbox access token initialization to main.js, fixes inconstent names, fixes Mapbox stylesheet inclusion in build process

danielfdsilva commented on README.md:
````diff
@@ -14,6 +14,14 @@ $ npm install
+```
+module.exports = {
+  api: 'http://localhost:4000',
+  mbToken: 'pk.eyJ1I...'
````
Since this is in the `production.js` which is committed, we can remove it from here

danielfdsilva commented on app/index.html:
```diff
@@ -42,14 +42,16 @@
 
     <!-- build:css /assets/styles/main.css -->
     <link rel="stylesheet" href="/assets/styles/main.css">
+    <link rel="stylesheet" href="/node_modules/mapbox-gl/dist/mapbox-gl.css">
```
This was actually my bad.
Can we move these 2 to the vendor build? lines 40-41

nbumbarger committed:
Update README.md, move vendor style link from main build to vendor in index

danielfdsilva committed:
Merge branch 'develop' into feature/map-component
